### ISSUE TITLE
[no-standalone-expect] allow specifying what to consider a test block via an option

### ISSUE THREAD

**Issue by G-Rath** [jest-community/eslint-plugin-jest#354]
This is pretty much the equivalent of `expect-expect`'s `assertFunctionNames` option: We should support consuming an array of arbitrary strings of which we should consider as "test blocks".

This would allow extensions to work nicely with this rule, such as [`jest-each`](https://github.com/jest-community/eslint-plugin-jest/pull/350#issuecomment-515569004).

It's probably best to just to make the default value the current blocks we test for, in the same way `expect-expect` has `expect` as it's default.

**Comment by yairkukielka** [comment]
If you're using jest-each and jest >=22, see [this](https://github.com/jest-community/eslint-plugin-jest/pull/350#issuecomment-516064696) answer.

**Comment by Lusito** [comment]
I'd like to see this too, since I've extended the describe.each and it.each functions with some additional properties.

**Comment by zwang180** [comment]
Just run into a situation where this would be really helpful: 

In the project that I'm working on, we are leveraging Next.js's environment variables feature to control the app's behavior, and we are trying to create unit tests coverage for those behaviors. Essentially we are doing something like:
```javascript
// test-utils.js
export const testIfA = process.env.MY_VAR === 'A' ? test : test.skip;
export const testIfB = process.env.MY_VAR === 'B' ? test : test.skip;
// others...

// myCode.test.js
import { testIfA, testIfB } from './test-utils.js';

describe('My tests', () => {
  testIfA('Only when A', () => {
    // test A behavior
  });

  testIfB('Only when B', () => {
    // test B behavior
  });
}); 
```

And `no-standalone-expect` will error out for any `expect` usage in both the `testIfA` and the `testIfB` block since it cannot recognize that it is a test block. Hence it will be really helpful if we could whitelist our helper functions to be considered as test blocks.
